src/ResolverUtils.js

```diff
@@ -28,18 +73,54 @@ export function getRouteMatches(match) {
   }));
 }
 
+function getRouteValue(match, getGetter, getValue) {
+  const { route } = match;
+  const getter = getGetter(route);
+  return getter ? getter.call(route, match) : getValue(route);
+}
+
 // This should work better with Flow than the obvious solution with keys.
 export function getRouteValues(routeMatches, getGetter, getValue) {
-  return routeMatches.map(match => {
-    const { route } = match;
-    const getter = getGetter(route);
-    return getter ? getter.call(route, match) : getValue(route);
-  });
+  return routeMatches.map(match => getRouteValue(match, getGetter, getValue));
+}
+
+/**
+ * Generate route data according to their getters, respecting the order of
+ * promises per the `defer` flag on routes.
+ */
+export function getRouteData(routeMatches, getGetter, getValue) {
+  return accumulateRouteValues(
+    routeMatches,
+    routeMatches[0].routeIndices,
+    ({ prevParentPromise, prevPromise }, match) => {
+      const { defer } = match.route;
+
+      // For a deferred route, the parent promise is the previous promise.
+      // Otherwise, it's the previous parent promise.
+      const parentPromise = defer ? prevPromise : prevParentPromise;
+
+      // If there is a parent promise, execute after it resolves.
+      const routeData = parentPromise
+        ? parentPromise.then(() => getRouteValue(match, getGetter, getValue))
+        : getRouteValue(match, getGetter, getValue);
+
+      return {
+        routeData,
+        prevPromise: isPromise(routeData) ? routeData : prevPromise,
+        prevParentPromise: parentPromise,
```

Comment:
@NSLS This is a small behavior change. It seemed odd for `defer` to change the parent promise route based on whether there was a previous promise. This makes it so that `defer` without any previous promises is a no-op, rather than making the children of that route with `defer` behave as if they all had `defer`.